src/redux/Mission/missionSlice.js

```diff
@@ -1,13 +1,57 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
 
-const initialState = [0];
+export const getMissions = createAsyncThunk('getMission', async () => {
```

Comment:
To avoid multiple calls to the API it is highly recommended to add some checks to see if this function was called previously or not. Maybe you can use
```
export const getMissions = createAsyncThunk('getMission', async (_, thunkAPI) => {
  if( thunkAPI.getState().missions,length !== 0) {
    return null
  }
```
Another solution would be to tweak the state of the application and add a state variable that is incremented whenever the function is called. If that variable is greater than 1 then you can simply return null before calling `axios` to get data from API.

Reply:
Thank you. This has also been fixed.